invokeai/backend/ip_adapter/ip_adapter.py

```diff
@@ -192,24 +198,46 @@ def _init_image_proj_model(self, state_dict):
         ).to(self.device, dtype=self.dtype)
 
 
+def load_ip_adapter_tensors(ip_adapter_ckpt_path: str, device: str) -> IPAdapterStateDict:
+    state_dict: IPAdapterStateDict = {"ip_adapter": {}, "image_proj": {}}
+
+    if ip_adapter_ckpt_path.endswith("safetensors"):
+        model = safe_open(ip_adapter_ckpt_path, device=device, framework="pt")
+        for key in model.keys():
+            if key.startswith("image_proj."):
+                state_dict["image_proj"][key.replace("image_proj.", "")] = model.get_tensor(key)
+            elif key.startswith("ip_adapter."):
+                state_dict["ip_adapter"][key.replace("ip_adapter.", "")] = model.get_tensor(key)
```

Comment:
Add an `else` condition so that we fail loudly if we try to load a model that does not match our assumptions.
```suggestion
            if key.startswith("image_proj."):
                state_dict["image_proj"][key.replace("image_proj.", "")] = model.get_tensor(key)
            elif key.startswith("ip_adapter."):
                state_dict["ip_adapter"][key.replace("ip_adapter.", "")] = model.get_tensor(key)
            else:
                raise RuntimeError(f"Encountered unexpected IP Adapter state dict key: '{key}'.")
```

Reply:
Do we need to? Because as long as the IP Adapter model has image projection tensors and adapter tensors, we can load it in as a valid model. Any extra keys that might exist in a future model would probably require a custom projection model anyway for the extra functionality but the core aspect of the IP Adapter should still be valid otherwise -- albeit it won't work like advertised but should still boot in.

I see why you want it added though. It would allow us to tightly control the state_dict for any fail case scenario on a custom model but I feel leaving this open here might be better for models that have lingering keys that are not deal breakers for the model itself.

Reply:
I feel like it's better to fail loudly if we receive unexpected keys. It means that either there is a bug in our handling code, or that there is a new feature that we need to add support for. I don't see any advantage to quietly running with an incomplete model - this just seems like it's setting us up for hard-to-debug "Why does it look bad?" issues.